addons/mail/models/mail_thread.py

```diff
@@ -267,7 +267,22 @@ def create(self, vals_list):
             for thread in threads:
                 subtype = thread._creation_subtype()
                 if subtype:  # if we have a subtype, post message to notify users from _message_auto_subscribe
-                    thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
+                    relations = subtype._get_auto_subscription_subtypes(self._name)[4]
+                    body = ''
+                    if relations:
```

Comment:
hum, we have too many indentation here, can we do something like this ?
```python
if not subtype:
    threads_no_subtype += thread
    continue
if not relations:
    continue

model_name = self.env['ir.model']._get(self._name).display_name
parent_models = ...
```

so we gain 2 indentations for the main part of the code

Reply:
Thanks, changed for subtype but unfortunately it was not possible to remove the "if relations" because there is a statement to execute after the "if" if the condition is False: send the mail with an empty body.